src/base.ts

```diff
@@ -41,76 +30,79 @@ const withPagination = <T>(json: {
   return { content, fieldsStats, pagination, topListing }
 }
 
-export const resolveServiceUrl = (service: Service): string => {
-  let url: string
-  switch (service) {
-    case Service.CAR:
-      url = apiClient.configuration.carServiceUrl
-      break
-    case Service.CATALOGUE:
-      url = apiClient.configuration.catalogueServiceUrl
-      break
-    case Service.SEARCH:
-      url = apiClient.configuration.searchServiceUrl
-      break
-    case Service.DEALER:
-      url = apiClient.configuration.dealerServiceUrl
-      break
-    case Service.OPTION:
-      url = apiClient.configuration.optionServiceUrl
-      break
-    case Service.ANALYTICS:
-      url = apiClient.configuration.analyticsServiceUrl
-      break
-    case Service.USER_NOTIFICATION:
-      url = apiClient.configuration.userNotificationServiceUrl
-      break
-    case Service.TOKEN_REFRESH:
-      url = apiClient.configuration.tokenRefreshServiceUrl
-      break
-    default:
-      throw new Error(`Tried to resolve url of unknown service "${service}"`)
+const authorizationHeader = () => {
+  if (!apiClient.tokens.accessToken) {
+    return null
   }
 
-  if ((url || "").trim()) {
-    return url
-  }
+  return `Bearer ${apiClient.tokens.accessToken}`
+}
 
-  throw new Error(`Missing endpoint configuration for "${service}" service`)
+export interface ApiCallOptions extends Omit<RequestInit, "method" | "body"> {
+  recaptchaToken?: string
+  headers?: Record<string, string>
+  host?: string
 }
 
-const authorizationHeader = () => {
-  if (!apiClient.tokens.accessToken) {
-    return null
+type Method = "GET" | "POST" | "PUT" | "PATCH" | "DELETE"
+
+interface RequestOptions extends ApiCallOptions {
+  method?: Method
+}
+
+export const getHost = (host: string = null) => {
+  if (!apiClient.configuration.host) {
+    throw new Error(
+      'ApiClient not configured, please run: ApiClient.configure({ apiGatewayUrl: "your.api.gateway" }'
+    )
   }
 
-  return `Bearer ${apiClient.tokens.accessToken}`
+  return host || apiClient.configuration.host
 }
 
-export const fetchPath = async (
-  service: Service,
-  path: string,
-  options = {}
-) => {
-  const prefix = resolveServiceUrl(service)
-  const url = `${prefix}/${stripLeadingSlash(path)}`
+const buildHeaders = ({
+  headers = {},
+  recaptchaToken,
+}: RequestOptions): Record<string, string> => {
+  const auth = authorizationHeader()
+
+  return {
+    "Content-Type": "application/json",
+    Accept: `application/vnd.carforyou.${apiClient.version}+json`,
+    ...(auth ? { Authorization: auth } : {}),
+    ...(recaptchaToken ? { "Recaptcha-Token": recaptchaToken } : {}),
+    ...headers,
+  }
+}
+
+export const fetchPath = async ({
+  path,
+  body,
+  options,
+}: {
+  path: string
+  body?: string
+  options: RequestOptions
+}) => {
+  const {
+    method = "GET",
+    host = null,
+    headers,
+    recaptchaToken,
```

Comment:
We could just create a helper where we use it and pass it as options.headers (Up to you would assume this to be a rather rare case just for form requests)

Reply:
I like passing it directly and having the client taking care of the header logic. This way the header name is local to the client and we don't have to worry about that. An additional benefit of lifting it up to the main wrapper is that we can now pass the recaptcha on any request should we need to.

Reply:
Ahh I did not ment to pass it form the consumer rather form the endpoint function as only a couple of endpoints will use it no strong opinion, up to you...

Reply:
Yeah, that's what I initially thought and then it ended up being a lot of endpoints that all manually transformed header so I lifted it.